platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ConfigurationCacheServices.kt

```diff
@@ -132,6 +132,8 @@ class ConfigurationCacheServices : AbstractGradleModuleServices() {
     object TaskExecutionAccessCheckerProvider : ServiceRegistrationProvider {
         @Provides
         fun createTaskExecutionAccessChecker(
+            /** In non-CC builds, [BuildTreeConfigurationCache] is not registered; accepting a list here is a way to ignore its absence. */
```

Comment:
💡 It would be nice to model this as an `Optional` dependency instead (cc @alllex 😉 )